packages/lisk-p2p/src/peer.ts

```diff
@@ -193,11 +193,17 @@ export class Peer extends EventEmitter {
 		return this._nodeInfo;
 	}
 
-	public connect(): void {
-		if (!this._outboundSocket) {
-			this._outboundSocket = this._createOutboundSocket();
-		}
-		this._outboundSocket.connect();
+	public async connect(): Promise<void> {
+
+		return new Promise<void>(resolve => {
+			if (!this._outboundSocket) {
+				this._outboundSocket = this._createOutboundSocket();
+			}
+			this._outboundSocket.connect();
+			this._outboundSocket.on('connect', _ => {
+				resolve();
```

Comment:
reject case?

Reply:
it will be done in a separate PR as Peer.connect needs a proper implementation.